filters/awssigner/awssigv4/sigv4.go

```diff
@@ -0,0 +1,193 @@
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	disableHeaderHoistingStr, ok := args[2].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	disableURIPathEscapingStr, ok := args[3].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	disableSessionTokenStr, ok := args[4].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	disableHeaderHoisting, err := strconv.ParseBool(disableHeaderHoistingStr)
+	if err != nil {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	disableURIPathEscaping, err := strconv.ParseBool(disableURIPathEscapingStr)
+	if err != nil {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	disableSessionToken, err := strconv.ParseBool(disableSessionTokenStr)
+	if err != nil {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	return &awsSigV4Filter{
+		region:                 region,
+		service:                service,
+		disableHeaderHoisting:  disableHeaderHoisting,
+		disableURIPathEscaping: disableURIPathEscaping,
+		disableSessionToken:    disableSessionToken,
+	}, nil
+}
+
+/*
+sigV4Filter is a request filter that signs the request.
+In case a non empty is body is present in request,
+the body is read and signed. The body is later reassigned to request. Operators should ensure
+that body size by all requests at any point of time is not more than the memory limit of skipper.
+*/
+func (f *awsSigV4Filter) Request(ctx filters.FilterContext) {
+	req := ctx.Request()
+
+	logger := log.WithContext(req.Context())
+
+	signer := NewSigner()
+
+	accessKey := getAndRemoveHeader(ctx, accessKeyHeader, req)
+	if accessKey == "" {
+		return
+	}
+
+	secretKey := getAndRemoveHeader(ctx, secretHeader, req)
+	if secretKey == "" {
+		return
+	}
+
+	sessionToken := getAndRemoveHeader(ctx, sessionHeader, req)
+	if sessionToken == "" {
+		return
+	}
+
+	timeStr := getAndRemoveHeader(ctx, timeHeader, req)
+	if timeStr == "" {
+		return
+	}
+
+	time, err := time.Parse(time.RFC3339, timeStr) //2012-11-01T22:08:41+00:00
+
+	if err != nil {
+		logger.Log(log.ErrorLevel, "time was not in RFC3339 format")
+		return
+	}
+
+	hashedBody, body, err := hashRequest(ctx, req.Body)
+	if err != nil {
+		logger.Log(log.ErrorLevel, fmt.Sprintf("error occured while hashing the body %s", err.Error()))
+		return
+	}
+	creds := internal.Credentials{
+		AccessKeyID:     accessKey,
+		SessionToken:    sessionToken,
+		SecretAccessKey: secretKey,
+	}
+
+	optfn := func(options *SignerOptions) {
+		options.DisableHeaderHoisting = f.disableHeaderHoisting
+		options.DisableSessionToken = f.disableSessionToken
+		options.DisableURIPathEscaping = f.disableSessionToken
+		options.Ctx = ctx.Request().Context()
+	}
+	//modifies request inplace
+	signer.SignHTTP(creds, req, hashedBody, f.service, f.region, time, optfn)
+
+	ctx.Request().Body = io.NopCloser(body) //ATTN: custom close() and read() set by skipper or previous filters are lost
```

Comment:
This should be also written in the user documentation: docs/reference/filters.md
All body filters that tap the body on request path, are impaired. This is fine but should be clear.

Reply:
Had missed this part. Added the documentation in latest commit, Thanks